### DISCUSSION TITLE
Custom UserDB Pydantic schema bug

### DISCUSSION THREAD

**Discussion by Ae-Mc** [fastapi-users/fastapi-users#873] [Q&A] [answered]
## Describe the bug

If I add column that will be generated on database side I can't add new user.
UserDB schema is exactly the same as User database model except default value.
But I don't want to define same default constraints in both models and schemas.

The exception is with Pydantic model validation.
The only non-workaround solution that I came up with is to create separate UserDBCreate model that would have same fields as UserCreate + hashed_password.

## To Reproduce

Just define Pydantic user schemas like this:
```python
class User(models.BaseUser):
    """User fetch pydantic model"""

    created_at: datetime


class UserCreate(models.BaseUserCreate):
    """User's creation pydantic scheme"""


class UserUpdate(models.BaseUserUpdate):
    """User's update pydantic scheme"""


class UserDB(User, models.BaseUserDB):
    """User stored in DB pydantic model"""

```

And table models like this (I use SQLAlchemy adapter):
```python
class User(Base, SQLAlchemyBaseUserTable):
    """Table for storing users"""

    created_at = Column(DateTime, default=datetime.now, nullable=False)
```

Than if you try to create user you will get validation error from Pydantic.

## Expected behavior

Successful user creation with automatically set created_at.

## Configuration
 - Python version : 3.10.1
 - FastAPI version : 0.71.0
 - FastAPI Users version : 9.2.2

### FastAPI Users configuration

Standard configuration from docs

**Comment by frankie567** [comment, marked as answer]
I wouldn't call it a bug in this case: it's more of a consequence of how FastAPI works.

In this context, the default value should be provided by the Pydantic model, not SQLAlchemy. Here is what I usually do in my projects:

```py
def now_utc():
    return datetime.now(timezone.utc)


class CreatedAt(BaseModel):
    created_at: datetime = Field(default_factory=now_utc)


class User(models.BaseUser, CreatedUpdatedAt):
    pass
```

The default is here handled automatically by Pydantic using the `default_factory` argument.
